test/irmin-pack/common.ml

```diff
@@ -269,3 +269,19 @@ module Filename = struct
           | None -> ""
           | Some f -> if stderr = stdout then " 2>&1" else " 2>" ^ quote f)
 end
+
+let check_string ~msg ~expected ~got =
+  let got = Hex.of_string got |> Hex.show in
+  Alcotest.(check string) (Fmt.str "%s" msg) expected got
+
+let check_iter iter_type (iter : 'a -> (string -> unit) -> unit) v checks =
+  let counter = ref 0 in
+  iter v (fun x ->
+      match List.nth_opt checks !counter with
+      | None -> Alcotest.failf "No more calls to %s left" iter_type
+      | Some (msg, expected) ->
+          let msg = Fmt.str "Check %s:%s" iter_type msg in
+          check_string ~msg ~expected ~got:x;
+          incr counter);
+  if !counter <> List.length checks then
+    Alcotest.failf "More calls to %s expected" iter_type
```

Comment:
These seem a bit too specific to me to go in `common`. Can we shift them to `test_hashes`?

Reply:
good point, done